client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/transaction/XaModeTransactionTable.java

```diff
@@ -392,13 +359,15 @@ private void onePhaseCommit() throws XAException {
             log.tracef("Registering remote cache '%s' for transaction xid=%s", remoteCache.getName(), currentXid);
          }
          return new TransactionContext<>(remoteCache.keyMarshaller(), remoteCache.valueMarshaller(),
-               remoteCache.getOperationsFactory(), remoteCache.getName());
+               remoteCache.getOperationsFactory(), remoteCache.getName(), remoteCache.isRecoveryEnabled());
       }
 
       private void cleanup() {
          registeredTransactions.remove(transaction);
-         registeredCaches.clear();
-         preparedCaches.clear();
+         //this instance can be used for recovery. we need at least one cache registered in order to access
```

Comment:
Do TMs really reuse XAResources like this? I would have thought it's more likely to create a new XAResource and call `recover()` on it directly.

Anyway, I think we should move the tx operations to a `GlobalOperationsFactory` that doesn't require a cache.

Reply:
create a new XaResource from where? There is no interfaces in the specifications to create XaResources, but they can be reused in other transactions (that's why we have Xid as parameters everywhere).

I don't see a `GlobalOperationFactory` but I really need one for transactions and counters...

Reply:
The ResourceAdapter seems to be an interface used by EE connector. And the Hot Rod client isn't one. Should it be? I'm not sure...
I was aware of XAResourceRecovery but it is related only to naryana. And, it looks like naryana likes to serialize and persist the xa-resources when possible. But I'm not sure if we can serialize ours...
The recovery process is really vague. Everything refers to XaResource.recover() but nothing says how to create them :(

Reply:
I found some more detail here, again Narayana-specific:

http://narayana.io/docs/product/index.html#d0e1032

Yeah, they like to serialize XAResources :) We could probably do it, we'd only need to name the cache managers and register them somewhere (e.g. JNDI) to allow deserialization.

I *think* we need to be a connector to implement crash recovery properly without TM-specific stuff, but I don't even want to investigate how much work that would mean right now :)